### ISSUE TITLE
Regression - current widget break juffed build

### ISSUE THREAD

**Issue by agaida** [lxqt/qtermwidget#131]
might be introduced in lxde/qtermwidget/pull/120 -- sorry, noticed it not earlier, didn't touch juffed for 274 days :( 

```
CMake Warning (dev) in /usr/share/cmake/qtermwidget5/qtermwidget5-config.cmake:
  Policy CMP0011 is not set: Included scripts do automatic cmake_policy PUSH
  and POP.  Run "cmake --help-policy CMP0011" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  The included script

    /usr/share/cmake/qtermwidget5/qtermwidget5-config.cmake

  affects policy settings.  CMake is implying the NO_POLICY_SCOPE option for
  compatibility, so the effects are applied to the including context.
Call Stack (most recent call first):
  plugins/terminal/CMakeLists.txt:6 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

Later this lead to  - no clue what i should change :) Sorry for the dir naming - i have the name schema $project snapshot upstream $features and fucked for builds that FTBFS.

```
/home/agaida/work/siduction/code/pkg-main/juffed/fucked/plugins/terminal/TerminalPlugin.cpp:10:25: fatal error: qtermwidget.h: No such file or directory
 #include "qtermwidget.h"
                         ^
compilation terminated.
plugins/terminal/CMakeFiles/terminal.dir/build.make:69: recipe for target 'plugins/terminal/CMakeFiles/terminal.dir/TerminalPlugin.cpp.o' failed
make[3]: *** [plugins/terminal/CMakeFiles/terminal.dir/TerminalPlugin.cpp.o] Error 1
```

EDIT: the nice thing about is - the new widget works with juffed, but i cant compile with the new widget - and have no clue, what to change.

**Comment by yan12125** [comment]
The build process can be fixed with the following patch:
```Diff
diff --git a/plugins/terminal/CMakeLists.txt b/plugins/terminal/CMakeLists.txt
index cfcd156..fcbf75e 100644
--- a/plugins/terminal/CMakeLists.txt
+++ b/plugins/terminal/CMakeLists.txt
@@ -10,9 +10,7 @@ else (USE_QT5)
     qt4_wrap_ui(terminal_UI_SRC ${terminal_UI})
 endif (USE_QT5)

-include_directories(${QTERMWIDGET_INCLUDE_DIRS})
-
 add_library(terminal SHARED ${terminal_SRC} ${terminal_MOC} ${terminal_UI_SRC})
-target_link_libraries(terminal ${JUFFED_LIBRARY} ${QTERMWIDGET_QT_LIBRARIES} ${QTERMWIDGET_LIBRARIES})
+target_link_libraries(terminal ${JUFFED_LIBRARY} qtermwidget5)
 install(TARGETS terminal DESTINATION ${JUFFED_PLUGINS_DIR})
```
That reveals a downside of lxde/qtermwidget#120 - it introduces a new usage, and the old way does not work anymore. All CMake-based projects relying on qtermwidget needs update. IMO it's a good direction to move to CMake's imported targets. The new scheme allows simpler ```CMakeLists.txt```. What we need is just a big BACKWARD INCOMPATIBILITY WARNING in the changelog.

**Comment by agaida** [comment]
There are not so much projects the use qtermwidget - i think qgis is using it, but these guys can read *and* write. So a big fat warning might be enough. For juffed i think i will only apply a patch and try to get this upstream - but the upstream is almost dead.

**Comment by agaida** [comment]
Cool, works fine, thank you very much

**Comment by abucodonosor** [comment]
@luis-pereira 

https://github.com/F1ash/qt-virt-manager/issues/27

howto fix that ?

**Comment by abucodonosor** [comment]
@luis-pereira 

with your changes ( without to have some backward compatibility) ,
if one wishes to support <0.8 and >= 0.8 versions you are very much
stuffed. So the only sane way is to s/ find_package()/pkg_check_modules()/
in your build files. Why ? well <0.8 returns 'unknow' version from find_package()
>0.8 ofc are now fine but you'll need add lots ifdef's and check stuff for nothing , right ?

Can you tell me where the problem is adding backward compatibility ?

**Comment by yan12125** [comment]
Here's a fix for qt-virt-manager:
```Diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index c6649ab..2175178 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1109,14 +1109,13 @@ include_directories(
     ${Spice1_INCLUDE_DIRS}
     ${SPICE_CLIENT_GLIB2_INCLUDE_DIRS}
     ${LibVirt_INCLUDE_DIRS}
-    ${QTERMWIDGET_INCLUDE_DIRS}
     ${LIBVNCCLIENT_INCLUDE_DIR}
 )
 set (VIEWER_BUILD_PROJECT_LIBRARIES
           ${BUILD_PROJECT_LIBRARIES})
 list(APPEND BUILD_PROJECT_LIBRARIES
     ${LibVirt_LIBRARIES}
-    ${QTERMWIDGET_LIBRARIES}
+    qtermwidget5
     ${GLib2_LIBRARIES}
     ${GOBJECT_LIBRARIES}
     ${GIO_LIBRARIES}
```
This fixes building. I didn't test the built binary as I don't know how qtermwidget is used in qt-virt-manager.

**Comment by agaida** [comment]
@yan12125 - thank you, should we place this as example in the README? Dunno which projects use qtermwidget - and if, in which way.

**Comment by yan12125** [comment]
An example in README is definitely helpful ;)

**Comment by abucodonosor** [comment]
@yan12125 

that work with >= 0.8 .. I figured that already , but if you want to support <0.8
too thats not the fix :)

and while at at you may want to fix the pyqt bindings these broke too

**Comment by agaida** [comment]
@abucodonosor - as discussed already, we are short on developers and developer time - so any pull requests for python would be helpful. And i guess that we don't want to support < 0.8 anymore - Qt 4 included.

**Comment by abucodonosor** [comment]
@agaida 

yes sure you but external packages are used in LTS distros also ..
What I mean is with that fix only peoples with >0.8 will be able to use
newer qt-virt-manager , someone with still <0.8 can't.

I've found an one liner using  pkg_check_modules() , that even works for Qt4..

Also I looked @pyqt however right now is a bit complicated since
qtermwidget_export.h is created on config time in $build folder
and qtermwidget.h inlcude it and .sip file include it from ../lib/. ,
qtermwidget_export.h doesn't exists at that point..

Maybe some sort version checking is needed an include from system 
however I've no idea how you guys want this to work.

**Comment by agaida** [comment]
To be true - i don't get it. Esp. the LTS part - LTS means imho some kind of "dead" code - so for LTS everything should be fine, they could use 0.7.x - the things become worse if we talk about backports - and for backports there is this magic thing called quilt as ultima ratio. I might be wrong.
